### PR TITLE
Optimizing brace expansion

### PR DESCRIPTION
**Pull request by deniszh**
According to @iain-buclaw-sociomantic comments in https://github.com/graphite-project/graphite-web/pull/1713/files `expand_braces()` is not really optimized.
I tested that using `timeit` - https://gist.github.com/deniszh/1e7eb64cb372d83d9ecd46d11c813193
According to tests, optimized version is 20% faster, so, it's worth trying.

Also change indentation back to 2 spaces.

### CODE REVIEW

File: webapp/graphite/finders/__init__.py [hunk 1]
```diff
@@ -42,37 +43,33 @@ def extract_variants(pattern):
...
-          res.extend(expand_braces(s[:open_brace] + sub.replace('}', '\\}') + s[close_brace + 1:]))
+def expand_braces(s):
+  res = list()
+
+  m = EXPAND_BRACES_RE.search(s)
+  if m is not None:
+    sub = m.group(1)
+    open_brace = s.find(sub)
+    close_brace = open_brace + len(sub) - 1
+    if sub.find(',') != -1:
```
Comment (iain-buclaw-sociomantic) [inline webapp/graphite/finders/__init__.py]: Alternatively, `if "," in sub:`. Whichever reads better I suppose. :-)

File: webapp/graphite/finders/__init__.py [hunk 2]
```diff
@@ -42,37 +43,33 @@ def extract_variants(pattern):
...
-      else:
-          res.extend(expand_braces(s[:open_brace] + sub.replace('}', '\\}') + s[close_brace + 1:]))
+def expand_braces(s):
+  res = list()
+
+  m = EXPAND_BRACES_RE.search(s)
+  if m is not None:
+    sub = m.group(1)
+    open_brace = s.find(sub)
+    close_brace = open_brace + len(sub) - 1
```
Comment (deniszh) [inline webapp/graphite/finders/__init__.py]: Yes, that's better

File: webapp/graphite/finders/__init__.py (hunk 1 above)
Comment (deniszh) [inline webapp/graphite/finders/__init__.py]: You mean, `if not "," in sub:`.
Not sure which one is better. :)

File: webapp/graphite/finders/__init__.py (hunk 2 above)
Comment (deniszh) [inline webapp/graphite/finders/__init__.py]: No, `open_brace, close_brace = m.span(1)` it's not the same. :(

File: webapp/graphite/finders/__init__.py (hunk 1 above)
Comment (deniszh) [inline webapp/graphite/finders/__init__.py]: Oh, I'm blind. Of course, `if "," in sub:`, and it's much better.

File: webapp/graphite/finders/__init__.py (hunk 2 above)
Comment (iain-buclaw-sociomantic) [inline webapp/graphite/finders/__init__.py]: Well, you have the following below:
```
      for pat in sub.strip('{}').split(','):
        res.extend(expand_braces(s[:open_brace] + pat + s[close_brace + 1:]))
    else:
        res.extend(expand_braces(s[:open_brace] + sub.replace('}', '\\}') + s[close_brace + 1:]))
```

With `m.span(1)` you can drop the ` + 1` bit.
```
      for pat in sub.strip('{}').split(','):
        res.extend(expand_braces(s[:open_brace] + pat + s[close_brace:]))
    else:
        res.extend(expand_braces(s[:open_brace] + sub.replace('}', '\\}') + s[close_brace:]))
```
:-)

Comment (deniszh) [inline webapp/graphite/finders/__init__.py]: Ah, in that sense. Yes, sorry, didn't get your point from start. Yes, it works, and more readable now.

**Comment by iain-buclaw-sociomantic** [comment]
Looks OK to me.

Following the linked bug report, I found the following revision change that this should probably be attributed to.
https://hg.python.org/cpython/rev/ec897bb38708

I was surprised to see how similar this now looks to it after my suggestions. :-)

**Comment by deniszh** [comment]
Wow, indeed, @iain-buclaw-sociomantic !
Thanks for suggestions!

**Comment by deniszh** [comment]
Merging for master and 0.9.x

**Pull request merged**